### PR TITLE
Fix issue with colliding atom types in structure addition

### LINKED ISSUE #1197: Wrong TIP3P water parameters saved out by ParmEd
Hi,

I tried to parametrize my ligand (69G.sdf) using OpenFF 1.2.1 and the protein (protein.pdb) using AMBER14SB and built the simulation box (and added TIP3P water + ions). I used ParmEd to save *gro and *top files out. However, I found in the *top file (complex.top), the water hydrogen LJ parameters were not correct:

```
[ atomtypes ]
; name    at.num    mass    charge ptype  sigma      epsilon
H4             1   1.007947  0.00000000  A     0.26495328      0.0656888
``` 
After checking the topology file, I found these hydrogen parameters were the same as some hydrogens on the ligand.  
For a TIP3P water, the H atoms should have following parameters:
```
[ atomtypes ]
; name    at.num    mass    charge ptype  sigma      epsilon
H6             1   1.007947  0.00000000  A              1              0
```

I also tried to build the system without the ligand and I did not have this issue. So I think ParmEd got confused by some hydrogen atoms on the ligand and mistakenly assigned these ligand hydrogen atom type to the water hydrogens when saving the gromacs topology out. Do you have any insights?

I attached my input files and scripts as well. 

Here is a list of package versions I used. 

```
openeye-toolkits          2020.1.0                 py37_0    openeye
openforcefield            0.7.1              pyh39e3cac_0    omnia
openforcefields           1.3.0                      py_0    omnia
parmed                    3.4.1            py37hcd2ae1e_0    conda-forge
openmm                    7.5.1            py37h96c4ddf_0    conda-forge
```

Thanks!

[input_scripts.zip](https://github.com/ParmEd/ParmEd/files/7188223/input_scripts.zip)

### PR DESCRIPTION
**Pull request by swails**
Also updates to OpenMM 7.6 support (previously only ported to v3.4
branch) and some other code cleanups.

Fixes #1197

### CODE REVIEW

File: parmed/tools/simulations/openmm.py [hunk 1]
```diff
@@ -16,7 +16,10 @@
 import sys
 import warnings
 try:
-    from simtk.openmm.vec3 import Vec3
+    try:
+        from simtk.openmm.vec3 import Vec3
+    except ImportError:
+        from openmm.vec3 import Vec3
```
Comment (peastman) [inline parmed/tools/simulations/openmm.py:22]: I suggest reversing this, and likewise in similar imports.  First try importing without `simtk`, then import with it if that fails.  That way you won't get a deprecation warning when using 7.6.

Comment (swails) [reply parmed/tools/simulations/openmm.py]: Ha, I actually just dropped support for OpenMM <7.6 altogether.

I'll hopefully finish up the CHARMM work soon so I can actually release ParmEd 4.0.

**Comment by swails** [comment]
This now also closes #1173

**Comment by swails** [comment]
I’m going to merge this as the test case verifies the bug is fixed.

**Comment by davidlmobley** [comment]
One of my folks just ran into this "bug" (issue/limitation) again; any chance we can get a release? It looks like this is only in master and not in a release yet.

**Comment by swails** [comment]
Yea I can do a 4.0 release soon. I merged a bunch of work into master recently.

**Pull request merged**